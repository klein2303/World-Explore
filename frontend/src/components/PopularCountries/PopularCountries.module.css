/* General CSS here */
.popularCountriesSection {
    display: flex;
    flex-direction: column;
    align-items: center;
    margin-top: 20px;
    margin-bottom: 60px;
}

.countryLabel {
    flex-direction: row;
    padding: 10px 20px;
    margin: 20px;
    font-size: 20px;
    background-color: #ffffff;
    color: #333;
    transition:
        background-color 0.3s,
        color 0.3s;
    cursor: pointer;
    border-radius: 10px;
}

.countryLabel:hover {
    background-color: #e4e1f7;
}

.popularTitle {
    margin: 20px;
    font-weight: 400;
    color: #424242;
    margin-bottom: 15px;
}

/*Display specific CSS below*/

@media (min-width: 1025px) {
    /*Display */
    .popularTitle {
        font-size: 20px;
    }

    .labelContainer {
        display: flex;
        flex-wrap: wrap;
        justify-content: center;
        max-width: 1000px;
    }

    .popularCountriesSection {
        margin-top: 15px;
    }
}

@media (max-width: 1024px) {
    /*Tablet */
    .popularTitle {
        font-size: 20px;
    }

    .labelContainer {
        display: flex;
        flex-wrap: wrap;
        justify-content: center;
        max-width: 80%;
        margin-bottom: 10%;
    }

    .countryLabel {
        margin: 5px;
    }

    .popularCountriesSection {
        margin-top: 0px;
    }
}

@media (max-width: 500px) {
    /* Phone*/
    .popularTitle {
        font-size: 17px;
    }

    .labelContainer {
        display: flex;
        flex-wrap: wrap;
        justify-content: center;
        max-width: 400px;
    }

    .countryLabel {
        flex-direction: row;
        padding: 10px 12px;
        margin: 7px;
        font-size: 14px;
        border-radius: 10px;
        border: 2px solid #d3d3d3;
        background-color: #ffffff;
        color: #333;
        transition:
            background-color 0.3s,
            color 0.3s;
    }
}
