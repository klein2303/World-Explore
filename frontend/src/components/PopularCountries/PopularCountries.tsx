import styles from "./PopularCountries.module.css";

const PopularCountries = () => {
    return (
        <section className={styles.popularCountriesSection} aria-label="Buttons with 10 most popular countries">
            <p className={styles.popularTitle} aria-label="Title">
                Popular countries
            </p>
            <section className={styles.labelContainer} aria-label="Labels for each country">
                {[
                    "Norway",
                    "USA",
                    "Finland",
                    "Madagascar",
                    "Germany",
                    "China",
                    "Sri lanka",
                    "Indonesia",
                    "Japan",
                    "Thailand",
                ].map((country, index) => (
                    <label key={index} className={styles.countryLabel} aria-label="label for country">
                        {country}
                    </label>
                ))}
            </section>
        </section>
    );
};

export default PopularCountries;
