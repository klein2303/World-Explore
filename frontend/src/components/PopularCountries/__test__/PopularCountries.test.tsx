import {render, screen} from "@testing-library/react";
import "@testing-library/jest-dom";
import {describe, it, expect} from "vitest";
import PopularCountries from "../PopularCountries";

describe("PopularCountries Tests", () => {
    it("renders the popular countries section correctly", () => {
        // Render the PopularCountries component
        const {asFragment} = render(<PopularCountries />);

        // Take a snapshot of the initial render
        expect(asFragment()).toMatchSnapshot();

        // Check if the section title is in the document
        expect(screen.getByLabelText("Title")).toHaveTextContent("Popular countries");

        // Check if the buttons for each country are in the document
        const countries = [
            "Norway",
            "USA",
            "Finland",
            "Madagascar",
            "Germany",
            "China",
            "Sri lanka",
            "Indonesia",
            "Japan",
            "Thailand",
        ];
        countries.forEach((country) => {
            expect(screen.getByText(country)).toBeInTheDocument();
        });
    });

    it("checks if country buttons are clickable", () => {
        render(<PopularCountries />);

        // Check that all buttons are present
        const buttons = screen.getAllByRole("button");
        expect(buttons.length).toBe(10); // Ensure there are 10 buttons

        // Check click behavior
        buttons.forEach((button) => {
            expect(button).toBeInTheDocument();

        });
    });
});
