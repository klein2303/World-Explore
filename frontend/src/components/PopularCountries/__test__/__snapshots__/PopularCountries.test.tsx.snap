// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PopularCountries Tests > renders the popular countries section correctly 1`] = `
<DocumentFragment>
  <section
    aria-label="Buttons with 10 most popular countries"
    class="_popularCountriesSection_f12b3d"
  >
    <p
      aria-label="Title"
      class="_popularTitle_f12b3d"
    >
      Popular countries
    </p>
    <section
      aria-label="Buttons for each country"
      class="_buttonContainer_f12b3d"
    >
      <button>
        Norway
      </button>
      <button>
        USA
      </button>
      <button>
        Finland
      </button>
      <button>
        Madagascar
      </button>
      <button>
        Germany
      </button>
      <button>
        China
      </button>
      <button>
        Sri lanka
      </button>
      <button>
        Indonesia
      </button>
      <button>
        Japan
      </button>
      <button>
        Thailand
      </button>
    </section>
  </section>
</DocumentFragment>
`;
